app/assets/javascripts/mapbox.coffee

```diff
@@ -387,12 +387,14 @@ update_all_option = (elem) ->
   return if $.inArray(elem.prop('id'), ['provider', 'stats_provider', 'zip_code', 'census_code']) < 0
   selected_elem_value = $("#selected_#{elem.prop('id')}").val().split(',')
 
-  if $.inArray('all', selected_elem_value) >= 0
+  new_selected = elem.val()
+
+  if ($.inArray('all', selected_elem_value) != -1 ||
+     $.inArray('all', new_selected) != -1) && new_selected != null
```

Comment:
Pure semantics, but I keep chained logic statements on one line.
  if ( $.inArray('all', selected_elem_value) != -1 || $.inArray('all', new_selected) != -1 ) && 
    new_selected != null

OR we could structure it a bit more
if new_selected != null && 
    ( $.inArray('all', selected_elem_value) != -1 ||
    $.inArray('all', new_selected) != -1 )

Reply:
Is the code actually wrong? This looks like a stylistic objection.

Reply:
If I don't wrap at the `||` the line is 91 characters long, which is passed my target of ~80.

Reply:
Mostly stylistic objection. I suggest putting the "new_selected != null" logic statement first for readability.
if new_selected != null &&
( $.inArray('all', selected_elem_value) != -1 ||
$.inArray('all', new_selected) != -1 )